addons/event_sale/models/event_registration.py

```diff
@@ -60,6 +74,19 @@ def _synchronize_so_line_values(self, so_line):
             }
         return {}
 
+    def _sale_order_ticket_type_change_notify(self, new_ticket_id):
+        for registration in self:
+            render_context = {
+                'registration': registration,
+                'old_ticket': registration.event_ticket_id.name,
+                'new_ticket': new_ticket_id.name
+            }
+            registration.sale_order_id.activity_schedule_with_view('mail.mail_activity_data_warning',
+                 user_id=registration.event_id.user_id.id or registration.sale_order_id.user_id.id or
+                         self.env.user.id if not self.env.user._is_public() else self.env.ref("base.user_admin").id,
```

Comment:
Wow.

Make a bunch of if/elif above to determine a user_id instead to ease readability.